node/src/main/scala/coop/rchain/node/runtime/NodeRuntime.scala

```diff
@@ -204,10 +198,18 @@ class NodeRuntime[F[_]: Monixable: ConcurrentEffect: Parallel: Timer: ContextShi
           proposerStateRefOpt,
           blockProcessor,
           blockProcessorState,
-          blockProcessorQueue
+          blockProcessorQueue,
+          routingMessageQueue
         )
       }
-      _ <- handleUnrecoverableErrors(program)
+
+      // Run main program and casper launch concurrently
+      messageProcess = Stream.eval(casperLaunch)
+      mainProgram    = Stream.eval(handleUnrecoverableErrors(program))
+      delay          = Stream.eval(Timer[F].sleep(3.seconds))
+
+      // TODO: implement cancellation with proper shutdown
+      _ <- (mainProgram concurrently (delay ++ messageProcess)).compile.drain
```

Comment:
This delay is temporary just to prevent sending messages until network is started.